### PR TITLE
Add account credentials

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- visit /account
- you should should see a list of your credentials (if not, open the deployment flow and add some)
- click Remove next to a credential
- it should get removed

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6924/

File: jujugui/static/gui/src/app/components/account/credentials/credentials.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+      });
+    },
+
+    /**
+      Retrieve the list of clouds.
+
+      @method _getClouds
+    */
+    _getClouds: function() {
+      const xhr = this.props.listClouds((error, clouds) => {
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/account/credentials/credentials.js]: Set loading to true here.

File: jujugui/static/gui/src/app/components/account/credentials/credentials.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      Handle deleting a credential.
+
+      @method _handleDeleteCredential
+      @param credential {String} A credential id.
+    */
+    _handleDeleteCredential: function(credential) {
+      const xhr = this.props.revokeCloudCredential(credential, (error) => {
+        if (error) {
+          const message = 'Unable to revoke the cloud credential';
+          this.props.addNotification({
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/account/credentials/credentials.js]: Thanks for pulling in addNotification and doing something more with the error than just logging to the console.

File: jujugui/static/gui/src/app/views/utils.js [hunk 3]
```diff
@@ -1808,6 +1808,9 @@ YUI.add('juju-view-utils', function(Y) {
         title: 'Local'
       }
     };
+    if (providerName === 'aws') {
```
Comment (kadams54) [inline jujugui/static/gui/src/app/views/utils.js]: Add a comment about why this is necessary.

**Comment by kadams54** [comment]
QA is :ok:!

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6925/

File: jujugui/static/gui/src/app/components/account/account.js [hunk 4]
```diff
@@ -25,10 +25,32 @@ YUI.add('account', function() {
...
+    */
+    _generatePaymentDetails: function() {
+      if (this.props.showPay) {
+        return (
+          <juju.components.AccountPaymentMethod
+            acl={this.props.acl}
+            addNotification={this.props.addNotification}
+            getUser={this.props.getUser}
+            username={this.props.userInfo.profile} />);
+      }
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/account/account.js]: This should return `null` as we had discussed - unless that causes issues with it rendering "null"

File: jujugui/static/gui/src/app/components/account/credentials/credentials.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+        </div>
+      );
+    }
+
+  });
+
+}, '', {
+  requires: [
+    'account-payment-method-card',
+    'expanding-row',
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/account/credentials/credentials.js]: I don't think that the `account-payment-method-card` and `expanding-row` components are used here.

File: jujugui/static/gui/src/app/views/utils.js [hunk 6]
```diff
@@ -1808,6 +1808,10 @@ YUI.add('juju-view-utils', function(Y) {
         title: 'Local'
       }
     };
+    // Sometimes the cloud id is ec2, othertimes aws, so map to the object key.
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/utils.js]: When it it sometimes?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6926/

**Comment by huwshimi** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**